R/get_transformers.R

```diff
@@ -47,33 +47,90 @@ get_transformers_flat <- function(strict = TRUE,
 #'
 #' Similar to [get_transformers_flat()], but additionally, returns some
 #'   functions needed due the fact that styling is done in a nested way.
+#' @param scope The extend of manipulation. Can range from "none" (least
+#'   invasive) to "token (most invasive). See 'Details'. This argument is a
+#'   vector of length one.
 #' @param indent_by How many spaces of indention should be inserted after
 #'   operators such as '('.
 #' @inheritParams get_transformers_flat
+#' @details The following options for `scope` are available.
+#'
+#' * "none": Performs no transformation at all.
+#' * "spaces": Manipulates spacing between token on the same line.
+#' * "line_breaks": In addition to "spaces", this option also manipulates
+#'   line breaks.
+#' * "tokens": In addition to "line_breaks", this option also manipulates
+#'   tokens.
+#'
+#' As it becomes clear from this description, more invasive operations can only
+#' be performed if all less invasive operations are performed too.
 #' @family obtain transformers
 #' @importFrom purrr partial
 #' @export
-get_transformers_nested <- function(strict = TRUE,
-                                    indent_by = 2,
-                                    start_comments_with_one_space = FALSE) {
-  c(create_filler,
-    partial(indent_round, indent_by = indent_by),
-    partial(indent_curly, indent_by = indent_by),
-    partial(indent_op, indent_by = indent_by),
-    partial(indent_without_paren, indent_by = indent_by),
-    get_transformers_flat(strict, start_comments_with_one_space),
-    force_assignment_op,
-    resolve_semicolon,
-    remove_line_break_before_curly_opening,
-    remove_line_break_before_round_closing,
-    add_line_break_afer_curly_opening,
-    add_line_break_before_curly_closing,
-    add_line_break_after_pipe,
-    remove_space_after_unary_pm_nested,
-    set_space_before_comments,
-    set_space_between_levels,
-    add_brackets_in_pipe,
-    strip_eol_spaces,
+get_transformers_nested <- function(
+  scope = "tokens",
+  strict = TRUE,
+  indent_by = 2,
+  start_comments_with_one_space = scope) {
```

Comment:
yes, if you go down all the way. Should be boolean though and independent of scope, so I am going to change the default. In Roxygen comments, indention should not be removed I think